SimTracker/SiPhase2Digitizer/test/Phase2TrackerValidateDigi.cc

```diff
@@ -684,6 +695,66 @@ unsigned int Phase2TrackerValidateDigi::getSimTrackId(edm::Handle<edm::DetSetVec
   }
   return simTrkId;
 }
+void Phase2TrackerValidateDigi::fillOTBXInfo() {
+
+  const edm::DetSetVector<PixelDigiSimLink>* links = otSimLinkHandle_.product();
+
+  for (typename edm::DetSetVector<PixelDigiSimLink>::const_iterator DSViter = links->begin(); DSViter != links->end(); DSViter++) {
+    unsigned int rawid = DSViter->id; 
+    DetId detId(rawid);
+    if (DetId(detId).det() != DetId::Detector::Tracker) continue;
+    int layer = tTopoHandle_->getOTLayerNumber(rawid);
+    if (layer < 0) continue;
+    std::map<uint32_t, DigiMEs >::iterator pos = layerMEs.find(layer);
+    if (pos == layerMEs.end()) continue;
+    DigiMEs& local_mes = pos->second;
+    int tot_digi = 0;
+    std::map<int, float> bxMap;
+    for (typename edm::DetSet< PixelDigiSimLink >::const_iterator di = DSViter->begin(); di != DSViter->end(); di++) {
+      tot_digi++;
+      int bx = di->eventId().bunchCrossing();
+      std::map<int, float>::iterator ic = bxMap.find(bx);
+      if (ic == bxMap.end()) bxMap[bx] = 1.0;
+      else bxMap[bx] += 1.0;
+    }
+    for (auto & v : bxMap) {
+      if (tot_digi) {
+	local_mes.BunchXTimeBin->Fill(v.first, v.second);
+	local_mes.FractionOfOOTDigis->Fill(v.first,v.second/tot_digi);
+      }
+    }
+  }
+}
+void Phase2TrackerValidateDigi::fillITPixelBXInfo() {
+
+  const edm::DetSetVector<PixelDigiSimLink>* links = itPixelSimLinkHandle_.product();
+
+  for (typename edm::DetSetVector<PixelDigiSimLink>::const_iterator DSViter = links->begin(); DSViter != links->end(); DSViter++) {
+    unsigned int rawid = DSViter->id; 
+    DetId detId(rawid);
+    if (DetId(detId).det() != DetId::Detector::Tracker) continue;
+    int layer = tTopoHandle_->getITPixelLayerNumber(rawid);
```

Comment:
@suchandradutta -please get the product itself instead of the Handle and use it in the loops (as done elsewhere in this class)

Also, could you update the title of this PR to reflect which validation suite is being updated.Thanks